### PR TITLE
Replace `EnumeratedList` and `BulletList` with MyST `List`

### PR DESCRIPTION
**Pull request by aktech**
I still need to handle the DefList case, as it sometimes comes inside the `EnumeratedList` and `BulletList`.

### CODE REVIEW

**Comment by Carreau** [comment]
+1, I was working on the same replacement this morning, I can try to merge this with my other changes in tree.py.

File: papyri/myst_ast.py [hunk 1]
```diff
@@ -99,6 +99,26 @@ class MParagraph(Node):
     # data: Any
 
 
+@register(4052)
+class MList(Node):
+    type = "list"
+    children: List["ListContent"]
+    ordered: bool = False
+    # start: int
+    # spread: bool
```
Comment (Carreau) [inline papyri/myst_ast.py:108]: One questions is are spread and start necessary for myst to render ?

Comment (aktech) [reply papyri/myst_ast.py:108]: No, I don't think so, from what I have seen so far.

**Comment by Carreau** [comment]
No sounds good, sorry, I was offline and ou of the country.

**Pull request merged**